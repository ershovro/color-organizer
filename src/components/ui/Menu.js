import React from 'react';
import {NavLink} from "react-router-dom";
import '../../stylesheets/menu.scss';

const selectedStyle = {
   color: 'red'
};

const Menu = ({className, sort}) => (
   <div className={`menu ${className}`}>
      <NavLink
         to="/"
         style={sort === '/' ? selectedStyle : {}}
         className="menu__link">
         date
      </NavLink>
      <NavLink
         to="/sort/title"
         activeStyle={selectedStyle}
         className="menu__link">
         title
      </NavLink>
      <NavLink
         to="/sort/rating"
         activeStyle={selectedStyle}
         className="menu__link">
         rating
      </NavLink>
   </div>
);


export default Menu;