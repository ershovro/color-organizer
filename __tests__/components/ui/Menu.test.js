import React from 'react';
import enzyme from 'enzyme';
import Menu from "../../../src/components/ui/Menu";
import { JSDOM } from 'jsdom';
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
import Adapter from 'enzyme-adapter-react-16';


const { window } = new JSDOM(`...`);
const { document } = (new JSDOM(`...`)).window;

enzyme.configure({ adapter: new Adapter() });
global.window = window;
global.document = document;

describe('<Menu /> ui component', () => {

   describe('Rendering ui', () => {

      it('Renders correctly', () => {
         compose(expect, toJSON, enzyme.shallow)( <Menu sort="/" /> ).
         toMatchSnapshot();
      });

      it('Renders default a div with the class .menu', () => {
         expect(
            enzyme.shallow( <Menu /> ).
            find('div.menu').length
         ).toBe(1);
      });

      it('Renders default 3 menu-item - a with the class menu__link', () => {
         expect(
            enzyme.shallow( <Menu /> ).
            find('.menu__link').length
         ).toBe(3);
      });

   });

});