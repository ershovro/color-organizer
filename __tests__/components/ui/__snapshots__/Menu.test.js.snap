// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> ui component Rendering ui Renders correctly 1`] = `
<div
  className="menu undefined"
>
  <NavLink
    className="menu__link"
    style={
      Object {
        "color": "red",
      }
    }
    to="/"
  >
    date
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "red",
      }
    }
    className="menu__link"
    to="/sort/title"
  >
    title
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "red",
      }
    }
    className="menu__link"
    to="/sort/rating"
  >
    rating
  </NavLink>
</div>
`;
